curator-recipes/src/test/java/org/apache/curator/framework/recipes/nodes/TestPersistentNode.java

```diff
@@ -136,4 +138,52 @@ public void testQuickCloseNodeExists() throws Exception
             CloseableUtils.closeQuietly(client);
         }
     }
+
+    @Test
+    public void testInterruption() throws Exception
```

Comment:
We'll need that otherwise I don't see the issue. The thread interrupt status is boolean. No matter how many times you set it it's still the same value.